### DISCUSSION TITLE
Multiple Sources at once

### DISCUSSION THREAD

**Discussion by yoshimo** [NightscoutFoundation/xDrip#2123] [General]
When you try to find the best product and compare accuracy , could that work with xdrip?
Add a libre2 and a libre3 and a g6 and g7 and an eversense as well as an accu chek device for a bloody baseline value and get curves in different colors basically :)
Other scenario could be that you have one sensor ending soon and you start a second sensor early to reduce the gap during the warmup time.
Maybe you also want to compare your own glucose levels to that of a friend or family member which you receive over an online connection.

**Comment by Navid200** [comment]
I could be wrong about this as I don't have any personal experience.  That's why I will be tagging others below.

I don't believe one xDrip installation can show readings from more than one source.  Can you imagine the complexity required to have multiple alert families?  Or can you imagine how confusing the alerts would be if readings from both sources triggered the same set of alerts?
 
But, you can install xDrip variants, on the same phone, and each can show readings from a different source.  
https://navid200.github.io/xDrip/docs/Variants.html

You could upload readings from each source to Nightscout.  Nightscout is very flexible.   I have seen Nightscout extensions that let you show readings on the same computer screen.  

With respect to starting a new sensor while the previous sensor is still active, it's a very bad idea with a device that offers factory calibration.  After you insert a G6 sensor, you should start it right away.  The longer you wait, the farther off your readings will be when you do start.
Unless you mean using two active transmitters simultaneously.  You can do that.  But, how many people have more than one active transmitter at a time?  I don't see a reason to invest developer time in something that can only serve an extremely small portion of the user community.

@jamorham @psonnera @tolot27

**Comment by tolot27** [comment]
As Navid200 mentioned, you can variants to test different sensors. You can also use two or more phones and upload the data to Nightscout and one follower phone which shows all the data.
Some time ago I weared L2 and G6 and measured my blood glucose with a glucometer every five minutes. The G6 outperformed L2 at my body. But this is could be different for others (personalized medicine).

**Comment by yoshimo** [comment]
Variants are more of a workaround. It is easier to compare curves that are shown side by side instead of different apps.
If you want to keep an eye on your children or spouse values as well as your own , a quick swipe on the graph to switch sources is easier than going to a second app. Also takes less space and energy to run.
It also is impossible to skip the warmup period that way by starting the replacement sensor early and then avoid the gap in the measurements by seamlessly going from one sensor to the next.

**Reply by Navid200** [reply]
Let's go one step at a time.
Let's cover the last point you have made first.
There is no way you can start a sensor while yo have the previous sensor still in action.

If you disagree, explain how you would do it with a single transmitter.

If you are talking about using two different transmitters, I don't think we are going to modify xDrip to support a minority.  The majority of users don't have more than one active transmitter at a time.  Otherwise, it would be very expensive.  
  
  
With respect to showing readings from more than one source, I don't think anyone debates that that can be useful.
However, there are serious technical implementation issues with that.  
Assume for a moment that we add this capability.  Do you want to get an alert when either person has a low without knowing which one is low?
This would be a disadvantage compared to variants.  If you use variants, you can choose identifiable tones so that you know as soon as the alert goes off who is low.

What is supposed to happen when using xDrip statistics?  You certainly wouldn't want the predicted A1C to take all the readings from both people into account.  So, we will need to provide two sets of statistics.  That's an example of the technical complexity this would bring.

How would we show history in this case?
When you go back to a day 30 days ago, do you expect to see both set of readings overlapped?  That wouldn't be very useful.  So, we will also need to provide two different sets of history records.
There is o end to this.  This will bring a lot of complexities with not much in return.

**Reply by yoshimo** [reply]
For a Libre2 the only thing that seems to prevent sensors overlapping is the official application.
I can pair my bluetooth headphone with my computer and my cell and the cellphone with the accu chek, my keys, the computer, the headphones. Multiple connections work fine on BT.

Looking at
http://jkaltes.byethost16.com/Juggluco/mgdL/index.html?i=1
it seems that the scenario is a valid one.

>  You can use multiple sensors beside each other; making it easy to move to a new sensor, without a gab between. If a sensor seemed unreliable you can add a sensor and see if that was warranted. 

You scan one sensor, link it up, 2 hours before the first one expires, you scan a second one. Both are paired , both send data , both can be scanned for spot values.

**Reply by yoshimo** [reply]
> Do you want to get an alert when either person has a low without knowing which one is low?

In times where i can set my messenger and phone apps to different alarms and ringtones for different contacts , this should be possible to solve.

**Reply by Navid200** [reply]
My mistake.  I thought you were talking about Dexcom G6.  Please ignore when I said you would need two transmitters.

**Reply by Navid200** [reply]
> > Do you want to get an alert when either person has a low without knowing which one is low?
> 
> In times where i can set my messenger and phone apps to different alarms and ringtones for different contacts , this should be possible to solve.

It may be possible.
There are lots of problems we need to solve.  We have serious bugs we need to solve.
So, we prioritize and move ahead solving what we can starting from the highest priority problems.
